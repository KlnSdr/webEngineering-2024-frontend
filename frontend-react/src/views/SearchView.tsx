import React, { useEffect, useState } from "react";
import AddProductLine from "../components/AddProductLine";
import Button from "react-bootstrap/Button";
import ListGroup from "react-bootstrap/ListGroup";
import Stack from "react-bootstrap/Stack";
import Heading, { Heading2 } from "../components/Heading";
import ImageArea from "../components/ImageArea";
import { NeededProduct, Product } from "../types/Products";
import {ProductsService} from "../services/ProductService";
import {RecipeService} from "../services/RecipeService";

function SearchView() {
  const [neededProducts, setNeededProducts] = useState<NeededProduct[]>([]);
  const [searchResults, setSearchResults] = useState<any[]>([]);
  const [productsData, setProductsData] = useState<Product[]>([]);

    //Fetch products from the backend
    useEffect(() => {
        ProductsService.getAll().then((products) => {
                setProductsData(products);
            });
    }, []);

}

export default SearchView;
