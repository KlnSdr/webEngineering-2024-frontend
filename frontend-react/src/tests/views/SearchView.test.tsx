import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import SearchView from "../../views/SearchView";

describe('SearchView', () => {
    it('renders correctly', () => {
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchView />
            </MemoryRouter>
        );

        // Check for static text elements
        expect(screen.getByText('Suche')).toBeInTheDocument();
        expect(screen.getByText('Produkte:')).toBeInTheDocument();
        expect(screen.getByText('Ergebnisse:')).toBeInTheDocument();

        // Check for buttons
        expect(screen.getByRole('button', { name: /Produkt hinzuf√ºgen/i })).toBeInTheDocument();
        expect(screen.getByRole('button', { name: /Suchen/i })).toBeInTheDocument();
    });

});
