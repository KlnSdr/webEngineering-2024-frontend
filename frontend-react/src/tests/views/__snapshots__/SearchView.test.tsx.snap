// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="search-view"
  >
    <h1
      class="headline-with-lines"
    >
      Suche
    </h1>
    <h2
      class="FoodTitle"
    >
      Produkte:
    </h2>
    <div
      class="col-md-4 mx-auto vstack gap-2"
    >
      <button
        class="bi bi-plus btn btn-primary"
        type="button"
      >
        Produkt hinzufügen
      </button>
      <button
        class="mt-3 bi bi-search btn btn-primary"
        type="button"
      >
        Suchen
      </button>
    </div>
    <h2
      class="FoodTitle"
    >
      Ergebnisse:
    </h2>
    <div
      class="list-group"
    >
      <div
        class="d-flex align-items-center list-group-item"
      >
        <span>
          Keine Rezepte gefunden
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-view"
  >
    <h1
      class="headline-with-lines"
    >
      Suche
    </h1>
    <h2
      class="FoodTitle"
    >
      Produkte:
    </h2>
    <div
      class="col-md-4 mx-auto vstack gap-2"
    >
      <button
        class="bi bi-plus btn btn-primary"
        type="button"
      >
        Produkt hinzufügen
      </button>
      <button
        class="mt-3 bi bi-search btn btn-primary"
        type="button"
      >
        Suchen
      </button>
    </div>
    <h2
      class="FoodTitle"
    >
      Ergebnisse:
    </h2>
    <div
      class="list-group"
    >
      <div
        class="d-flex align-items-center list-group-item"
      >
        <span>
          Keine Rezepte gefunden
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
